### ISSUE TITLE
How to view result after save/delete/update?

### ISSUE THREAD

**Issue by EmirGluhbegovic** [ethanresnick/json-api#163]
I want to send an email when a user is created.
How would I get that result of the create, update or delete in the server from JSON-API?

Regards,
Emir

**Comment by ethanresnick** [comment]
There are a few high-level ways to do this. One is to leverage mongoose middleware. The other is to use a query transform to intercept the adapter result in `query.returning`. The latter option looks something like:

```
app.get('/:type(people)',
  Front.transformedAPIRequest((req, query) => {
    if(!('addNameList' in req.query)) {
      return query;
    }

    const origReturning = query.returning;
    return query.resultsIn(async (...args) => {
      const origResult = await origReturning(...args);
      // read the created user from either `origResult.document.primary` or from
      // the `args`. Reading `origResult.document` is a bit indirect, but the structure
      // of `args` is the last thing that's still likely to change before v3 is finalized, so
      // reading from the origResult is a bit safer.
      return origResult;
    })
  })
);
```

**Comment by ethanresnick** [comment]
As a bit of an aside, I'm going to try to work on the documentation more once v3 is done, so I'm very curious to hear how you think it could be improved. In particular, I'm wondering: where did you have to dig into the source code to figure out something that was under-documented? What concepts were tricky, even after viewing the source? And, conversely, was there anything that was documented in excessive/needless detail in the README, where those details would've been better deferred to some secondary document or just left in the source?
